### PR TITLE
core: Don't offer implicit dependency results

### PR DESCRIPTION
**Pull request by Makman2**
Closes https://github.com/coala/coala/issues/4458

### CODE REVIEW

File: tests/core/CoreTest.py [hunk 1]
```diff
@@ -489,14 +493,16 @@ def test_run_complex(self):
             [(BearD_NeedsC.name, self.section1.name, self.filedict1)])
 
         # For BearE -> BearA.
-        bear_a = get_next_instance(BearA,
-                                   (result.bear for result in results))
+        bear_a = get_next_instance(
+            BearA,
+            (bear.bear for bear in bear_e.dependency_results[BearA]))
```
Comment (Makman2) [inline tests/core/CoreTest.py]: hm I should rather rename the first `bear` to `result`

Comment (Makman2) [reply tests/core/CoreTest.py]: Same for the other occurrences.

**Comment by Makman2** [comment]
unack a019d0c2f51e72940834f639eaef7cb6f986fb1b

File: tests/core/CoreTest.py (hunk 1 above)
Comment (palash25) [reply tests/core/CoreTest.py]: `bear.bear` does look a bit confusing :confused:

**Comment by Makman2** [comment]
Was already approved before and I only renamed some variables. Taken a look myself too again.

**Comment by Makman2** [comment]
ack 508ca49edd778dede052aef86de4051b2113af4a

**Comment by Makman2** [comment]
@gitmate-bot ff

**Pull request merged**